### PR TITLE
Semantic snippets: don't provide snippets when dotting of a type name

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Fixes: https://youtu.be/V_eRvgl5sVg?t=2190

### CODE REVIEW

**Comment by DoctorKrolic** [comment]
@akhera99 PTAL on this simple fix

**Comment by DoctorKrolic** [comment]
@MartyIX I don't think we have a centralized place where all capabilities of semantic snippets are described. Maybe take a look at [the original proposal](https://github.com/dotnet/roslyn/issues/56541), but it has quite a few outdated/not yet implemented details. But in short I can describe semantic snippets thre following way:
- They appear in completion list only where they are actually valid (e.g. you won't get `class` snippet inside a method)
- They try to avoid compiler errors when inserted (e.g. `class` snippet by default creates `MyClass` type, bu if there is already such name in the scope, snippet will generate `MyClass1` to avoid naming conflict)
- They respect your code style settings (e.g. if you have 'Require accessibility modifiers' setting set to 'always' then `class` snippet will generate `public class MyClass` instead of just `class MyClass`)
- They have additional advanced features that aren't available for legacy snippets. For instance, we have what we reffer to as "inline statement snippets". These can be inserted if you type them after `.` as a member access, which is very useful (I myself use this thing _a lot_). An example here is:
devenv_HMOWDZkxfA

**Review by akhera99 — APPROVED** [review]
Looks good, thanks!

File: src/Features/Core/Portable/Snippets/SnippetProviders/AbstractInlineStatementSnippetProvider.cs [hunk 1]
```diff
@@ -83,7 +83,19 @@ private static bool TryGetInlineExpressionInfo(SyntaxToken targetToken, ISyntaxF
...
         {
             var expression = syntaxFacts.GetExpressionOfMemberAccessExpression(parentNode)!;
+
             var typeInfo = semanticModel.GetTypeInfo(expression, cancellationToken);
+            var symbolInfo = semanticModel.GetSymbolInfo(expression, cancellationToken);
+
+            // If we get the same symbol from symbol info as expression's type
+            // then we are dotting from a type name itself, e.g. `string.$$`.
+            // Inline statement snippets are not valid in this context
+            if (SymbolEqualityComparer.Default.Equals(symbolInfo.Symbol, typeInfo.Type))
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Snippets/SnippetProviders/AbstractInlineStatementSnippetProvider.cs:93]: i'm wary about this due to the `Color Color` case.  Can we add a test or that.

also, to check for this case, i think you just need to check if the symbolInfo.Symbol is an INamedTypeSymbol (or just an ITypeSymbol).  You don't really need to check if they're equal.

**Comment by DoctorKrolic** [comment]
@CyrusNajmabadi PTAL

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
looks good.  thanks!

**Pull request merged**